Add policy for removal of products (#8834)
ComplianceAsCode/content: docs/manual/developer/01_introduction.md

- Keeping track of the decomposition: Metadata in product.yml files, CODEOWNERS file
- Area of effect: All PSPs that don’t have specific stakeholders, associated content and test scenarios.
- Guidelines: Care needs to be taken when content requires a slightly different approach for different products - don’t copy and paste, and also don’t change existing behavior for everybody.
  See the [style guide](04_style_guide.md) for more details.

##### Removal of Products

> Products that do not have any contributions for two years may be subject to removal.

Reviewer: ```suggestion
Products
  - that are past their declared EOL date, or
  - that don't have contributions to files in their respective product folder for two years may be subject to removal.
```

I suggest this formulation, so it is easier to track the two-years period, as one can add a prodtype to a rule, and that could also qualify as a "contribution". But how do you detect that? It is probably reasonable to expect that a product needs gradual profile updates if it is relevant.

The following process shall be used:

> 1. A pull request is created removing the product.

Reviewer: ```suggestion
1. The intention to remove the product is publicly communicated.
1. A pull request is created removing the product.
```

Creating a PR is quite difficult, so better to do it if we know that there will be no veto.

  - Past their declared end-of-life date
  - Have not had contributions to files in their respective product folder for **two years**

The following process shall be used to start the removal of the product:
1. A GitHub issue is created proposing the removal of the product.
    The issue must contain the following:
    1. The end date and time (in UTC) of the comment period 
    1. A mention of the last contributor to the product, more than one is preferred
    1. The reason for removal (lack of contributions or product EOL)
1. A comment period of **21 days** shall be observed.
The issue description must explicitly state the date and time (in UTC) when the comment period will close.
1. In the case of lack of contributions, if there are no new contributors that volunteer once the comment period closes the product shall be removed.
   Products that are going EOL the product shall be removed unless there is an objection form a representative of product.
The pull request removing the product should include the removal of
    1. The product folder in `products/`
    1. any Jinja templates that use the product

>     1. the product from all `prodtyes`

Reviewer 2: ```suggestion
    1. the product from all `prodtype`
```
